tensorboard/webapp/plugins/npmi/views/data_selection/metric_search/metric_search_component.ng.html

```diff
@@ -0,0 +1,42 @@
+<!--
+@license
+Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<mat-icon svgIcon="search_24px"></mat-icon>
+<input
+  matInput
+  #matInput
+  autocomplete="off"
+  placeholder="Add Metric Filter"
+  [value]="regexFilterValue"
+  (input)="onRegexFilterValueChange.emit($event.target.value)"
+  [matAutocomplete]="filterMatches"
+/>
+<mat-icon
+  svgIcon="error_24px"
+  *ngIf="!isRegexFilterValid"
+  class="error-icon"
+  matTooltip="Invalid regex filter. The result may be stale."
+>
+</mat-icon>
+<mat-autocomplete
+  #filterMatches="matAutocomplete"
+  autoActiveFirstOption
+  (optionSelected)="onAddFilter.emit($event.option.value); matInput.value=''"
```

Comment:
since `onAddFilter` will trigger a `this.store.dispatch(npmiActions.npmiMetricsRegexChanged({regex: ''}));` (container L116), is it necessary to call `matInput.value=''` here, or can it be dropped?

Reply:
It seems to be. Otherwise the text will remain on selection. I think this is a mat-autocomplete thing.

Reply:
After some investigation, it seems that pressing Enter when the input is empty fails (the option text remains in the input), because
- the before string === the after string, so the selector doesn't trigger an update
- Angular must be setting input.value directly, outside of change detection

I think it's reasonable to keep the call, but could we please
- move this into a callback function, say `onOptionSelected`
- add a comment in that function, saying that matInput.value needs to be cleared manually, since the Angular Material Component may be modifying the input value outside of change detection
?

Reply:
Done.